test/common/testlib.py

```diff
@@ -739,74 +744,6 @@ def run(self, result=None):
 
         self.currentResult = None
 
-    def setUp(self):
-        if opts.address and self.provision is not None:
-            raise unittest.SkipTest("Cannot provision multiple machines if a specific machine address is specified")
-
-        self.machine = None
-        self.browser = None
-        self.machines = {}
-        provision = self.provision or {'machine1': {}}
-
-        # First create all machines, wait for them later
-        for key in sorted(provision.keys()):
-            options = provision[key].copy()
-            if 'address' in options:
-                del options['address']
-            if 'dns' in options:
-                del options['dns']
-            if 'dhcp' in options:
-                del options['dhcp']
-            machine = self.new_machine(**options)
-            self.machines[key] = machine
-            if not self.machine:
-                self.machine = machine
-            if opts.trace:
-                print("Starting {0} {1}".format(key, machine.label))
-            machine.start()
-
-        # Now wait for the other machines to be up
-        for key in self.machines.keys():
-            machine = self.machines[key]
-            machine.wait_boot()
-            address = provision[key].get("address")
-            if address is not None:
-                machine.set_address(address)
-            dns = provision[key].get("dns")
-            if address or dns:
-                machine.set_dns(dns)
-            dhcp = provision[key].get("dhcp", False)
-            if dhcp:
-                machine.dhcp_server()
-
-        if self.machine:
-            self.journal_start = self.machine.journal_cursor()
-            self.browser = self.new_browser()
-        self.tmpdir = tempfile.mkdtemp()
-
-        def sitter():
-            if opts.sit and not self.checkSuccess():
-                if self._outcome:
-                    [traceback.print_exception(*e[1]) for e in self._outcome.errors if e[1]]
-                else:
-                    self.currentResult.printErrors()
-                sit(self.machines)
-        self.addCleanup(sitter)
-
-        def intercept():
-            if not self.checkSuccess():
-                self.snapshot("FAIL")
-                self.copy_js_log("FAIL")
-                self.copy_journal("FAIL")
-                self.copy_cores("FAIL")
-        self.addCleanup(intercept)
-
-    def tearDown(self):
-        if self.checkSuccess() and self.machine.ssh_reachable:
-            self.check_journal_messages()
-            self.check_browser_errors()
```

Comment:
I think sitter(), intercept(), and this part of the teardown (check journal and browser errors) should be part of `BaseCase`, to avoid redundancy.

Reply:
ack